snakemake/jobs.py

```diff
@@ -317,6 +317,9 @@ def conda_env_file(self):
         if self._conda_env_file is None:
             expanded_env = self.rule.expand_conda_env(self.wildcards_dict)
             if expanded_env is not None:
+                # insert parameters
+                expanded_env = expanded_env.format(params=self.params)
```

Comment:
I think this rather should happen in rule.expand_conda_env, which could take params as an argument, similar to how rule.expand_params takes resources etc. as additional arguments.

Reply:
Thanks a lot for the suggestion! I updated the code accordingly.